test/functional/CompositeContainerTest.php

```diff
@@ -337,4 +345,34 @@ public function testThreeLevelComplexRetrieval()
 
         $this->assertEquals($expected, $actual, 'The container structure did not resolve services correctly');
     }
+
+    /**
+     * Tests the method that creates a {@see NotFoundException}.
+     *
+     * @since [*next-version*]
+     */
+    public function testCreateNotFoundException()
+    {
+        $subject = $this->createInstance();
+
+        $expected = new NotFoundException('test message', 3, null);
+        $exception = $subject->this()->_createNotFoundException('test message', 3, null);
+
+        $this->assertEquals($expected, $exception);
```

Comment:
Go ahead. You just need to test `getMessage()`, `getCode()`, and `getPrevious()`.